### PR TITLE
Use Amazon Simple Email Service for outgoing email

### PR DESCRIPTION
**Pull request by eddierubeiz**
Ref #760 

Documentation is at https://chemheritage.atlassian.net/wiki/spaces/HDCSD/pages/1046151169/Amazon+Simple+Email+Server

Do not deploy before merging the Ansible staging branch to production ( the PR is at https://bitbucket.org/ChemicalHeritageFoundation/ansible-inventory/pull-requests/111/ )and ensuring the new variables are available in production. (That said, if you do, nothing bad will happen. But we want to do things in the correct order).

### CODE REVIEW

File: app/lib/scihist_digicoll/env.rb [hunk 1]
```diff
@@ -288,6 +284,18 @@ def self.shrine_dzi_storage
...
+    # From address:
+    define_key :no_reply_email_address
+    # To addresses (these are email lists maintained by IT.)
+    define_key :digital_tech_email_address
+    define_key :digital_email_address
+
+    # These are the credentials used for accessing Amazon's simple email server:
+    define_key :ses_smtp_username
+    define_key :ses_smtp_password
     define_key :smtp_host
```
Comment (jrochkind) [inline app/lib/scihist_digicoll/env.rb:299]: These values would apply even if it wasn't `ses`, right? Change variable names to just `smtp_username` and `smtp_password`, what do you think?

File: config/environments/production.rb [hunk 2]
```diff
@@ -90,12 +90,15 @@
...
       raise RuntimeError, "Please specify smtp_host in local_env.py so we can send emails."
     end
+
     config.action_mailer.delivery_method = :smtp
     config.action_mailer.smtp_settings = {
-      address: ScihistDigicoll::Env.lookup(:smtp_host),
-      domain: 'sciencehistory.org',
-      ssl: false,
-      enable_starttls_auto: false
+      :address => "email-smtp.us-east-1.amazonaws.com",
```
Comment (jrochkind) [inline config/environments/production.rb:96]: Don't hard-code this address in, use it from `ScihistDigicoll::Env`, there is even an `stmp_host` key already there. 

oops sorry just noticed this was marked "wip", I"m reviewing prematurely sorry!

File: app/lib/scihist_digicoll/env.rb (hunk 1 above)
Comment (eddierubeiz) [reply app/lib/scihist_digicoll/env.rb:299]: Absolutely, duh!

File: config/environments/production.rb (hunk 2 above)
Comment (eddierubeiz) [reply config/environments/production.rb]: Yup -- but I have fixed this. I also got rid of the vestigial `email_server_ip_dns` setting in both Ansible and scihist_digicoll.

**Comment by jrochkind** [comment]
@eddierubeiz 

1. Have you looked into the "getting SES approved for production traffic" situation? Please let us know here on PR ticket that you've confirmed this. 

2. Please confirm on PR ticket here that you've verified email being sent via SES using this code has been received. (I think it should work in dev even, if the Env is all set? Might be fun to test that out and verify). 

3. In dev or test WITHOUT the variables being set, what happens, and does it differ from what happened before this PR? Please investigate and let us know on this PR. 

Thanks!

**Comment by jrochkind** [comment]
Also can you link here to ansible PR url, so I can find it, and to make it easier to verify it's been merged? I'm not used to dealing with that ansible bitbucket, so have trouble finding things there. Thanks!

**Comment by eddierubeiz** [comment]
`1) Have you looked into the "getting SES approved for production traffic" situation? Please let us know here on PR  ticket that you've confirmed this, what it entailed, link to relevant SES docs etc.`

No, we are *not* yet approved for production traffic; we're only allowed to send up to 200 messages a day, at a rate of 1 per second. You can monitor our progress against this quota at https://console.aws.amazon.com/ses/home?region=us-east-1#dashboard. (I've already sent out a few test messages today, as you can see.)

Docs are at https://docs.aws.amazon.com/ses/latest/DeveloperGuide/request-production-access.html for getting us to production -- we just have to fill out a prose description of how we plan to use our new email powers for good, basically.

I can take care of this before we merge / deploy this PR, as promised.

**Comment by eddierubeiz** [comment]
`2. Please confirm on PR ticket here that you've verified email being sent via SES using this code has been received. (I think it should work in dev even, if the Env is all set? Might be fun to test that out and verify).`

Confirmed -- it works fine in staging. (You could test it in dev too, of course, but testing it in staging has the added advantage of being able to double-check it works correctly with the changes I've made to Ansible.)

**Comment by eddierubeiz** [comment]
`3. In dev or test WITHOUT the variables being set, what happens, and does it differ from what happened before this PR? Please investigate and let us know on this PR.`

In dev or test, if you don't have any mail variables set, ActionMailer::Base.mail( stuff ).deliver_now will fail silently, as it did before.

(Note that we are just changing the value of smtp_host and adding two new credentials to allow connections to SES: as you noted the other day, we're already using SMTP, so it's just a question of changing *which* server we're connecting to and making sure we're all set up.)

**Pull request merged**